_posts/2021-04-06-run-minitest-on-github-actions-with-parallel-jobs-using-build-matrix.md

```diff
@@ -0,0 +1,121 @@
+---
+layout: post
+title:  "Run Minitest on Github Actions with parallel jobs using build matrix"
+date:   2021-04-06 08:00:00 +0100
+author: "Artur Trzop"
+categories: continuous_integration
+og_image: "/images/blog/posts/run-minitest-on-github-actions-with-parallel-jobs-using-build-matrix/github-minitest.jpeg"
+---
+
+How to run Ruby on Rails tests in Minitest on Github Actions? What to do if tests are slow? How to manage complex workflows? You can use Github Actions build matrices to divide Minitest files between jobs and run the test suite much faster.
+
+<img src="/images/blog/posts/run-minitest-on-github-actions-with-parallel-jobs-using-build-matrix/github-minitest.jpeg" style="width:300px;margin-left: 15px;float:right;" alt="Minitest, Ruby, Github, Github Actions" />
+
+If your Minitest tests are taking dozen minutes and you would like to save some time for your Ruby engineering team then you could use tests parallelization on your CI server.
+
+To run tests as fast as possible you need to split them into equal buckets (into parallel jobs). But how to do it? Some of the test files can be super fast to execute, other Minitest files can take minutes if they run system tests (E2E tests).
+
+There is also an aspect of preparing the test environment for each parallel job. By preparing I mean you need to clone a repository, install ruby gems or load them from a cache, maybe you need to load some docker container, etc. This can take various times on each parallel job. Random network errors happen like network delay to [load cached gems](/2021/how-to-load-ruby-gems-from-cache-on-github-actions), or maybe Github Actions from time to time will start one of your jobs late comparing to others. It's an inevitable issue in the network environment and can cause your tests to run a different amount of time on each parallel job. This is visible on the graph below and it causes the CI build to be slower.
```

Comment:
`start one of your jobs late comparing to others`
-> `compared to others` or `relative to others`